packages/scripts/src/commands/app-data/report/reporters/asset-summary.ts

```diff
@@ -0,0 +1,185 @@
+
+interface IReportData {
+  path: string;
+  count: number;
+  size_kb?: number;
+  missing?: true;
+}
+
+interface IAssetsSummary extends IReportTable {
+  data: IReportData[];
+}
+
+/** Generate summary reports for assets */
+export class AssetsSummaryReport {
+  /** List of known extensions to search assets for */
+  private assetExtensions: string[] = [];
+
+  private reportSummary: Record<string, number> = {};
+
+  constructor(private assetData: IAssetEntryHashmap) {
+    this.generateAssetExtensionList(Object.keys(this.assetData));
+  }
+
+  public async process(sheetData: IParsedWorkbookData) {
+    // HACK - check for assets within all the content of all the flows
+    // Could be replaced with more fine-grained checks like example `extractTemplateAssets`
+    this.checkForAssets(sheetData);
+
+    // NOTE - could also consider checking for assets within the deployment config itself
+    // although as these are absolute paths and also may not be required by frontend.
+    // So for now just leave to authors judgement whether assets like android/icon.jpg
+    // are actually 'unused' or not from report
+
+    const summaryData: IReportData[] = Object.entries(sortJsonKeys(this.reportSummary)).map(
+      ([path, count]) => {
+        const entry: IReportData = { path, count };
+        const assetDataEntry = this.assetData[path];
+        if (assetDataEntry) {
+          entry.size_kb = assetDataEntry.size_kb;
+        } else {
+          entry.missing = true;
+        }
+        return entry;
+      }
+    );
+
+    // Reports
+
+    const assetSummaryData = summaryData.filter((v) => !v.missing);
+    const asset_summary: IAssetsSummary = {
+      data: assetSummaryData,
+      display: "collapse_closed",
+      title: "Matched Assets",
+      description:
+        "Assets that are used within sheets and also can be found in the synced asset data",
+      type: "table",
+      columns: ["path", "size_kb", "count"],
+    };
+
+    const assetMissingData = summaryData.filter((v) => v.missing);
+    const assets_missing: IReportTable = {
+      data: assetMissingData,
+      display: "collapse_open",
+      title: "Missing Assets",
+      description: "Assets that have references within sheets but do not appear in app-data",
+      type: "table",
+      columns: ["path", "count"],
+    };
+
+    const assetUnusedData = this.generateUnusedAssetsList();
+    const assets_unused: IReportTable = {
+      data: assetUnusedData,
+      display: "collapse_open",
+      title: "Unused Assets",
+      description: "Assets that appear in app-data but do not have references within sheets",
+      type: "table",
+      columns: ["path", "size_kb"],
+    };
+
+    const assets_size: IReportTable = {
+      data: this.generateAssetSizeList(summaryData, assetUnusedData),
+      display: "collapse_open",
+      title: "Asset Size",
+      description: "Assets that appear in app-data but do not have references within sheets",
```

Comment:
Description doesn't make sense. I think no description is needed, can just remove this line.

Reply:
Oh yeah, think it was copy-pasted from the previous one. Agreed no description required.

Reply:
Resolved with [8e562b7](https://github.com/IDEMSInternational/open-app-builder/pull/2429/commits/8e562b7935ebc0852baa722fc028f5c6d7250ec2)